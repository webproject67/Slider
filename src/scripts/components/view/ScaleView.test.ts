import ScaleView from './ScaleView';

const state = {
  min: 0,
  max: 100,
  from: 0,
  fromPercent: 0,
  to: 100,
  toPercent: 100,
  step: 1,
  view: false,
  range: false,
  flag: true,
  progress: true,
  scale: true,
};

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new ScaleView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    state.view = true;
    expect(new ScaleView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element3', () => {
    state.min = 10;
    state.max = 20;
    expect(new ScaleView(state).getElement()).toMatchSnapshot();
  });
});
