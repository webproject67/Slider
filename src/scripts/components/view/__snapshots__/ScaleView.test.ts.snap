// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__list"
>
  <div
    class="slider__item"
    style="left: 0%;"
  >
    |
    <div
      class="slider__quantity"
    >
      0
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 5%;"
  >
    |
    <div
      class="slider__quantity"
    >
      5
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 10%;"
  >
    |
    <div
      class="slider__quantity"
    >
      10
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 15%;"
  >
    |
    <div
      class="slider__quantity"
    >
      15
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 20%;"
  >
    |
    <div
      class="slider__quantity"
    >
      20
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 25%;"
  >
    |
    <div
      class="slider__quantity"
    >
      25
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 30%;"
  >
    |
    <div
      class="slider__quantity"
    >
      30
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 35%;"
  >
    |
    <div
      class="slider__quantity"
    >
      35
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 40%;"
  >
    |
    <div
      class="slider__quantity"
    >
      40
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 45%;"
  >
    |
    <div
      class="slider__quantity"
    >
      45
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 50%;"
  >
    |
    <div
      class="slider__quantity"
    >
      50
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 55%;"
  >
    |
    <div
      class="slider__quantity"
    >
      55
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 60%;"
  >
    |
    <div
      class="slider__quantity"
    >
      60
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 65%;"
  >
    |
    <div
      class="slider__quantity"
    >
      65
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 70%;"
  >
    |
    <div
      class="slider__quantity"
    >
      70
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 75%;"
  >
    |
    <div
      class="slider__quantity"
    >
      75
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 80%;"
  >
    |
    <div
      class="slider__quantity"
    >
      80
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 85%;"
  >
    |
    <div
      class="slider__quantity"
    >
      85
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 90%;"
  >
    |
    <div
      class="slider__quantity"
    >
      90
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 95%;"
  >
    |
    <div
      class="slider__quantity"
    >
      95
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 100%;"
  >
    |
    <div
      class="slider__quantity"
    >
      100
    </div>
  </div>
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__list slider__list_state_transformed"
>
  <div
    class="slider__item"
    style="left: 0%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      0
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 5%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      5
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 10%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      10
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 15%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      15
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 20%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      20
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 25%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      25
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 30%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      30
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 35%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      35
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 40%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      40
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 45%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      45
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 50%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      50
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 55%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      55
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 60%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      60
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 65%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      65
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 70%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      70
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 75%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      75
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 80%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      80
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 85%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      85
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 90%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      90
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 95%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      95
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 100%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      100
    </div>
  </div>
</div>
`;

exports[`snapshot renders correctly element3 1`] = `
<div
  class="slider__list slider__list_state_transformed"
>
  <div
    class="slider__item"
    style="left: 0%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      10
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 10%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      11
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 20%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      12
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 30%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      13
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 40%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      14
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 50%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      15
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 60%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      16
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 70%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      17
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 80%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      18
    </div>
  </div>
  <div
    class="slider__item"
    style="left: 90%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      19
    </div>
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="display: none;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    />
  </div>
  <div
    class="slider__item"
    style="left: 100%;"
  >
    |
    <div
      class="slider__quantity slider__quantity_state_transformed"
    >
      20
    </div>
  </div>
</div>
`;
